src/navigation/bottom-sheet/createBottomSheetNavigator.tsx

```diff
@@ -16,8 +16,14 @@ import type {
   BottomSheetNavigationOptions,
 } from './types';
 import BottomSheetNavigatorView from './views/BottomSheetNavigatorView';
+import { RootStackParamList } from '@/navigation/types';
 
-type Props = DefaultNavigatorOptions<BottomSheetNavigationOptions> &
+type Props = DefaultNavigatorOptions<
+  RootStackParamList,
+  StackNavigationState<RootStackParamList>,
+  BottomSheetNavigationOptions,
+  BottomSheetNavigationEventMap
+> &
```

Comment:
Our custom bottom sheet had a ton of TS errors, but I think I did OK. Some stuff lingers though 🤔